articles/ros2_security_enclaves.md

```diff
@@ -0,0 +1,315 @@
+Similarly, private and public keystore materials can also be pushed down into their own respective sub-folders within the root keystore directory.
+This is reminiscent of the pattern used earlier in Keymint [1].
+
+
+```
+$ tree keystore/
+keystore
+├── enclaves
+│   └── ...
+│       └── ...
+├── private
+│   ├── ca.csr.pem
+│   └── ca.key.pem
+└── public
+    ├── ca.cert.pem
+    ├── identity_ca.cert.pem -> ca.cert.pem
+    └── permissions_ca.cert.pem -> ca.cert.pem
+```
+
+
+### ``public``
+
+The ``public`` directory contains anything permissible as public, such as public certificates for the identity or permissions certificate authorities.
+As such, this can be given read access to all executables.
+Note that in the default case, both the `identity_ca` and `permissions_ca` points to the same CA certificate.
+
+### ``private``
+
+The ``private`` directory contains anything permissable as private, such as private key material for aforementioned certificate authorities.
+This directory should be redacted before deploying the keystore onto the target device/robot.
+
+### ``enclaves``
+
+The ``enclaves`` directory contains the security artifacts associated with individual enclaves, and thus node directories are no longer relevant.
+Similar to node directories however, the `enclaves` folder may still recursively nest sub-paths for organizing separate enclaves.
+The `ROS_SECURITY_ROOT_DIRECTORY` environment variable should by convention point to this directory.
+
+
+## Runtime
+
+Given the normative case where a enclave within a policy may be specific to a single node/container process, the namespace the node is remapped to will inevitably affect the required security permissions necessary within the enclave.
+To highlight this interdependency, and to help avoid enclave path collisions, a hierarchy borrowing namespaces is appropriate. 
+By convention, ros2launch could be used to prefix relative enclave paths for single process node or containers using the namespace in scope, to enable a convention of composable launch files with adjustable and parameterized enclave paths.
+Given the runtime command argument for specifying the fully qualified enclave path, ros2launch would accordingly resolve relative enclave paths for executables, as defined by launch attributes.
+
+### Unqualified enclave path
+
+For single process nodes with unqualified enclave paths, the enclave directory will subsequently default to the root level enclave.
+
+``` xml
+<launch>
+  <node pkg="demo_nodes_cpp" exec="talker"/>
+  <node pkg="demo_nodes_cpp" exec="listener"/>
+</launch>
+```
+
+```
+$ tree enclaves/
+enclaves/
+├── cert.pem
+├── governance.p7s
+├── identity_ca.cert.pem -> ../public/identity_ca.cert.pem
+├── key.pem
+├── permissions_ca.cert.pem -> ../public/permissions_ca.cert.pem
+└── permissions.p7s
+```
+
+### Pushed unqualified enclave path
```

Comment:
I think we should left all `launch` related discussion out of this document for now, as it's not clear how this would be implemented.

Reply:
Maybe we can keep the content but move it to a "Future/TBD" category ?
This version of the design doc should highlight this is uncertain and not implemented

Reply:
Done: 3e148b2